### ISSUE TITLE
ResponseFactoryInterface is not inter-operable with other libs

### ISSUE THREAD

**Issue by delboy1978uk** [http-interop/http-factory#36]
Since the namespace is Interop and not Psr, I can't use the ResponseFactory with my lib. Expects a Psr\blahblah, got an Interop one instead.

As you know, the PSR hasn't been fully approved as of yet, so there is no official package.

I got around this by defining `Psr\\Http\\Message\\` namespace to `src/`, creating a ResponseFactoryInterface.php and literally copying the interface from here:
https://github.com/php-fig/fig-standards/blob/master/proposed/http-factory/http-factory.md#21-requestfactoryinterface

Then I created a wrapper class extending `Http\Factory\Diactoros\ResponseFactory` and implementing `Psr\Http\Message\ResponseFactoryInterface`. Then it worked.

So basically, can you change the namespace for your interface until we have an official psr vendor lib?

**Comment by shadowhand** [comment]
There is no `Psr` interface because PSR-17 hasn't been accepted. What/Where are you seeing this requirement? Everything in `http-interop` is pointing back to HTTP Interop interfaces.

**Comment by delboy1978uk** [comment]
Hi @shadowhand, I'm trying to use your Factory but as you defined your own interfaces it's non interoperable with any other lib. I had to do this as a workaround in my tests https://github.com/delboy1978uk/middleware/blob/master/tests/unit/ResponseFactory.php

**Comment by shadowhand** [comment]
@delboy1978uk Why wouldn't you use `http-interop/http-factory-diactoros`? There is no PSR that defines these interfaces...

**Comment by delboy1978uk** [comment]
there isn't an accepted one, so we need to write those interfaces ourselves until there is a psr 17 composer lib. you did that, but changed the namespace, so it won't work with anyone else's psr-15 stuff

**Comment by delboy1978uk** [comment]
like so https://github.com/delboy1978uk/middleware/blob/master/src/ResponseFactoryInterface.php

**Comment by prisis** [comment]
http-interop/http-factory is the base repro for the the new psr-17 standard

The same was for https://github.com/container-interop/container-interop now it is the new psr standard

**Comment by danopz** [comment]
The problem with using the PSR interfaces from the current draft is that they may change over time until the final version is released. Doing so you would write implementations with your never updating interfaces and may don't get notified about changes.
Then you have interfaces like this which provides releases for the interfaces so you can follow the changes and are able to implement defined milestones. You just need to rename the namespaces after the PSR gets releases.

**Comment by delboy1978uk** [comment]
Yes, that's indeed true, but if you bother to implement a non accepted PSR interface, you'll end up swapping your interface out when it does become accepted, so where's the difference? We cant inter-operate your lib with other libs while you use your own namespace. and again, we can always git tag as a zero point something until it **is** accepted. If you don't agree with changing the namespace to Psr, you might as well close this issue. Personally, I would use the Psr namwespace, and tag the lib v0.x.x until we have an accepted interface.

**Comment by shadowhand** [comment]
https://github.com/php-fig/http-factory is now released. This package, as of version 0.4.1, extends the official interfaces.
